packages/manager/apps/pci-vouchers/src/components/vouchers/BuyCreditModal.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  OsdsButton,
+  OsdsFormField,
+  OsdsInput,
+  OsdsModal,
+  OsdsSpinner,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_INPUT_TYPE,
+  ODS_SPINNER_SIZE,
+  OdsInputValueChangeEvent,
+} from '@ovhcloud/ods-components';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+} from '@ovhcloud/ods-common-theming';
+
+import { useCallback, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useEnvironment } from '@ovh-ux/manager-react-shell-client';
+import { useBuyCredit } from '@/hooks/useVouchers';
+
+interface BuyCreditModalProps {
+  projectId: string;
+  onClose: () => void;
+  onSuccess: (amount: number, url: string) => void;
+  onError: (cause: Error) => void;
+}
+
+const isValidInputAmount = (amount: number) => {
```

Comment:
Directly inside the component:
`const isAmountValid = useMemo(
    () => Number.isNaN(amount) && amount >= 0 && amount < 100000000,
    [amount],
  );`